### ISSUE TITLE
[feature request] removing the need to use the `accelerate` launcher

### ISSUE THREAD

**Issue by stas00** [huggingface/accelerate#2585]
### System Info

```Shell
x
```

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

x

### Expected behavior

Sometimes there is a need to be able to launch an Accelerate-based program w/o the `accelerate` launcher - for example, in case once wants to use the super-optimized SLURM resource placements and
having `srun` launch all the individual per-gpu processes.

e.g. lightning doesn't use a launcher, e.g. see [here](https://github.com/stas00/ml-
engineering/blob/master/orchestration/slurm/launchers/lightning-
launcher.slurm) and here is how I can launch a normal torch.distributed multi-
gpu program w/o `torchrun`: [here](https://github.com/stas00/ml-
engineering/blob/master/orchestration/slurm/launchers/srun-launcher.slurm) \-
or more specifically it's just:  

    srun bash -c "RANK=\$SLURM_PROCID LOCAL_RANK=\$SLURM_LOCALID python my-dist-prog.py"

which to the program is just:  
    
    RANK=\$SLURM_PROCID LOCAL_RANK=\$SLURM_LOCALID python my-dist-prog.py

Of course, we want the advanced features supported - so Deepspeed/FSDP, etc.

And ideally, it'd be the same accelerate config file - so it can be re-used w/ and w/o the launcher.

Thank you.

**Comment by muellerzr** [comment]
To be clear:

This feature request is *not* to make `accelerate launch` optional at its core, it's to make it optional for *advanced* features using only a `config` file and *not* manually utilizing items like the `DeepSpeedPlugin` or `FSDPPlugin`.

**Comment by muellerzr** [comment]
I need to stew on this, as I can see a user getting very confused as to why `python myfile.py` and passing in an `accelerate_config_yaml` to the `Accelerator` doesn't magically spawn up DDP how they want.

I'd rather you just use the `DeepSpeedPlugin` or `FSDPPlugin` itself, as the API is currently.

**Comment by stas00** [comment]
yes, sure, what I meant is that ideally the `accelerate` launcher is there if the user wants to use it, but they could launch the program in other ways and Accelerate can figure out the rest just from the config file.

For example to use a Deepspeed-based program, you can use the `deepspeed` launcher or any other launcher, i.e. the launcher isn't part of the program itself. i.e. the 2 shouldn't depend on each other to work, other than the program needs whatever env vars passed for it to work. As an example `torch.dist` requiring `RANK`, `LOCAL_RANK`, `WORLD_SIZE` but it doesn't care if the user launched each process manually.

**Comment by stas00** [comment]
> I need to stew on this, as I can see a user getting very confused as to why python myfile.py and passing in an accelerate_config_yaml to the Accelerator doesn't magically spawn up DDP how they want.

You won't be passing accelerate_config_yaml to the program, that won't work. You'd be setting:

```
ACCELERATE_CONFIG_FILE=config.yaml RANK=... LOCAL_RANK=... python myprog.py
```

**Comment by muellerzr** [comment]
As an `env` var I'm far more comfortable with

**Comment by stas00** [comment]
As mentioned in my last comment - I think extending the `torch.distributed` requirements would be the most intuitive API, especially if it's just a single additional env var.

Perhaps the documentation could then show it in this exact context and then it's crystal clear to the user how it fits with the env vars they are already hopefully well familiar with.
